test/generators/UserAnswersEntryGenerators.scala

```diff
@@ -31,93 +30,93 @@ import queries.Gettable
 trait UserAnswersEntryGenerators {
   self: Generators =>
 
-  def generateAnswer: PartialFunction[Gettable[_], Gen[(QuestionPage[_], JsValue)]] =
+  def generateAnswer: PartialFunction[Gettable[_], Gen[JsValue]] =
     generatePreTaskListAnswer orElse
       generateTraderDetailsAnswer orElse
       generateGuaranteeDetailsAnswer
 
-  private def generatePreTaskListAnswer: PartialFunction[Gettable[_], Gen[(QuestionPage[_], JsValue)]] = {
-    case OfficeOfDeparturePage   => arbitrary[CustomsOffice](arbitraryOfficeOfDeparture).map(Json.toJson(_)).map((OfficeOfDeparturePage, _))
-    case ProcedureTypePage       => arbitrary[ProcedureType].map(Json.toJson(_)).map((ProcedureTypePage, _))
-    case DeclarationTypePage     => arbitrary[DeclarationType].map(Json.toJson(_)).map((DeclarationTypePage, _))
-    case TIRCarnetReferencePage  => Gen.alphaNumStr.map(JsString).map((TIRCarnetReferencePage, _))
-    case SecurityDetailsTypePage => arbitrary[SecurityDetailsType].map(Json.toJson(_)).map((SecurityDetailsTypePage, _))
-    case DetailsConfirmedPage    => Gen.const(true).map(JsBoolean).map((DetailsConfirmedPage, _))
+  private def generatePreTaskListAnswer: PartialFunction[Gettable[_], Gen[JsValue]] = {
```

Comment:
Hello Jack, can you please explain what has changed that you are then able to get rid of the QuestionPage type?

Reply:
This is migrating from the old style of building an arbitrary user answers to our new style. Since the input of the partial function is the page, it seems pointless to include it as part of the tuple in the output